python/ray/tests/test_actor_advanced.py

```diff
@@ -203,7 +203,7 @@ def test_actor_init_fails(ray_start_cluster_head):
     cluster = ray_start_cluster_head
     remote_node = cluster.add_node()
 
-    @ray.remote(max_restarts=1)
+    @ray.remote(max_restarts=1, max_task_retries=-1)
```

Comment:
Why do we need all the changes in the unit tests? It's a bit concerning because it seems to indicate there might be some compatibility issue with current user applications.

Reply:
All the modified Python and Java test cases are due to the assumption that actor tasks sent after the actor is failed will be cached and sent later when the actor is back alive. This is exactly what we are trying to change in this PR.

However, I think I can revert the addition of `max_task_retries=-1` and instead add a loop to check if the actor is back alive before sending subsequent tasks. This can reduce the use of `max_task_retries` in test cases and should be more readable and maintainable. This method is used in a Java test case because right now `max_task_retries` cannot be configured in Java and it's always 0.

Reply:
Got it, thanks! I think the tests are okay as is then, just wanted to make sure we don't have regressions here. I took a look at the [docs](https://docs.ray.io/en/latest/ray-core/actors/fault-tolerance.html) too, and I think this PR actually better matches what we currently state there, so seems fine to modify the tests to match that behavior.